packages/mdl-snackbar/foundation.js

```diff
@@ -0,0 +1,142 @@
+/**
+ * Copyright 2016 Google Inc. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {MDLFoundation} from 'mdl-base';
+import {cssClasses, strings} from './constants';
+
+export default class MDLSnackbarFoundation extends MDLFoundation {
+  static get cssClasses() {
+    return cssClasses;
+  }
+
+  static get strings() {
+    return strings;
+  }
+
+  static get defaultAdapter() {
+    return {
+      addClass: (/* className: string */) => {},
+      removeClass: (/* className: string */) => {},
+      hide: () => {},
+      unhide: () => {},
+      setMessageText: (/* message: string */) => {},
+      setActionText: (/* actionText: string */) => {},
+      hideAction: () => {},
+      unhideAction: () => {},
+      registerActionClickHandler: (/* handler: EventListener */) => {},
+      deregisterActionClickHandler: (/* handler: EventListener */) => {},
+      registerTransitionEndHandler: (/* handler: EventListener */) => {},
+      deregisterTransitionEndHandler: (/* handler: EventListener */) => {}
+    };
+  }
+
+  constructor(adapter) {
+    super(Object.assign(MDLSnackbarFoundation.defaultAdapter, adapter));
+
+    this.active = false;
+    this.queue_ = [];
+  }
+
+  init() {
+    this.actionClickHandler_ = () => this.invokeAction_();
+    this.adapter_.registerActionClickHandler(this.actionClickHandler_);
+    this.adapter_.hide();
+  }
+
+  destroy() {
+    this.adapter_.deregisterActionClickHandler(this.actionClickHandler_);
+  }
+
+  showSnackbar(data) {
+    if (data === undefined) {
```

Comment:
hmm, actually...I'm thinking maybe instead of just passing in an object, we pass in a `message` argument as the first parameter, and then an options object as a second parameter which could include information about actions.

```javascript
show(message, options = {}) {
  if (!message) {
    // throw
  } 
  if (data.actionHandler && !data.actionText) {
    // throw
  }
  // do other stuff
}
```

I also agree with @Garbee that we should probably just call `showSnackbar` `show()` instead.